tests/spin/test_lattice.py

```diff
@@ -0,0 +1,518 @@
+            ],
+        ),
+    ],
+)
+def test_boundary_condition(vectors, positions, n_cells, boundary_condition, expected_edges):
+    r"""Test that the correct edges are obtained for given boundary conditions"""
+    lattice = Lattice(
+        n_cells=n_cells, vectors=vectors, positions=positions, boundary_condition=boundary_condition
+    )
+    assert sorted(lattice.edges) == sorted(expected_edges)
+
+
+@pytest.mark.parametrize(
+    # expected_edges here were obtained manually.
+    ("vectors", "positions", "n_cells", "neighbour_order", "expected_edges"),
+    [
+        (
+            [[0, 1], [1, 0]],
+            [[0, 0]],
+            [3, 3],
+            2,
+            [
+                (0, 1, 0),
+                (1, 2, 0),
+                (3, 4, 0),
+                (4, 5, 0),
+                (6, 7, 0),
+                (7, 8, 0),
+                (0, 3, 0),
+                (3, 6, 0),
+                (1, 4, 0),
+                (4, 7, 0),
+                (2, 5, 0),
+                (5, 8, 0),
+                (2, 4, 1),
+                (1, 3, 1),
+                (5, 7, 1),
+                (4, 6, 1),
+                (0, 4, 1),
+                (1, 5, 1),
+                (3, 7, 1),
+                (4, 8, 1),
+            ],
+        ),
+        (
+            [[0, 1], [1, 0]],
+            [[0, 0]],
+            [3, 4],
+            3,
+            [
+                (0, 1, 0),
+                (1, 2, 0),
+                (2, 3, 0),
+                (4, 5, 0),
+                (5, 6, 0),
+                (6, 7, 0),
+                (8, 9, 0),
+                (9, 10, 0),
+                (10, 11, 0),
+                (0, 4, 0),
+                (4, 8, 0),
+                (1, 5, 0),
+                (5, 9, 0),
+                (2, 6, 0),
+                (6, 10, 0),
+                (3, 7, 0),
+                (7, 11, 0),
+                (0, 5, 1),
+                (4, 9, 1),
+                (1, 6, 1),
+                (5, 10, 1),
+                (2, 7, 1),
+                (6, 11, 1),
+                (1, 4, 1),
+                (5, 8, 1),
+                (2, 5, 1),
+                (6, 9, 1),
+                (3, 6, 1),
+                (7, 10, 1),
+                (0, 8, 2),
+                (1, 9, 2),
+                (2, 10, 2),
+                (3, 11, 2),
+                (0, 2, 2),
+                (4, 6, 2),
+                (8, 10, 2),
+                (1, 3, 2),
+                (5, 7, 2),
+                (9, 11, 2),
+            ],
+        ),
+        ([[1, 0], [0.5, np.sqrt(3) / 2]], [[0.5, 0.5 / 3**0.5], [1, 1 / 3**0.5]], [2, 2], 0, []),
+    ],
+)
+def test_neighbour_order(vectors, positions, n_cells, neighbour_order, expected_edges):
+    r"""Test that the correct edges are obtained for given neighbour order"""
+    lattice = Lattice(
+        n_cells=n_cells, vectors=vectors, positions=positions, neighbour_order=neighbour_order
+    )
+    assert sorted(lattice.edges) == sorted(expected_edges)
```

Comment:
What is the actual order `edges` are created? 🤔 I'm guessing, you might be using some kind of traversal in the `KDTree` for them. Eventually, it might be useful to mention the order somewhere in the docstring or as a comment.

Reply:
KDTree uses depth first search to locate the nearest neighbours (i.e. distance). Is that what you want to be added to the docstring?

Reply:
Yep! That should be enough.